### DISCUSSION TITLE
[Feature]: support rust plugin and rust loader

### DISCUSSION THREAD

**Discussion by cordmaker** [web-infra-dev/rspack#7816] [Announcements]
### What problem does this feature solve?
Allows developers to write custom, high-performance (compared to JS) plugins or loaders using the Rust language.
### What does the proposed API of configuration look like?

like farm

**Comment by hardfist** [comment]
it's not easy to make  the rust loader | plugin compatible in all platforms, so we're not going to support rust loader| plugin in the near future, 
but we will support rust api first, so if you need high performance loader & plugin implementation, you can try use rust api to build in the future
